src/Integration/ErrorListenerIntegration.php

```diff
@@ -53,15 +57,34 @@ public function setupOnce(): void
                 return;
             }
 
-            if ($exception instanceof SilencedErrorException && !$this->options->shouldCaptureSilencedErrors()) {
+            $currentHub = Hub::getCurrent();
+            $client = $currentHub->getClient();
+
+            // The client could have been detached from the hub. If this is the
+            // case this integration should not run
+            if (null === $client) {
+                return;
+            }
+
+            $integration = $client->getIntegration(self::class);
```

Comment:
Sure, I understood that. But I see that here you're getting the integration from the client...

Reply:
Yup this is how it should be done:
```
$currentHub = Hub::getCurrent();
$integration = $currentHub->getIntegration(self::class);
```

Reply:
It's the same, since `Hub::getCurrent()->getIntegration(...)` is really just a shortcut for `Hub::getCurrent()->getClient()->getIntegration(...)`

Reply:
Then we should consistently use always the same path, I would say from the Hub for shortness.

Reply:
It's the same, since `Hub::getCurrent()->getIntegration(...)` is really just a shortcut for `Hub::getCurrent()->getClient()->getIntegration(...)`. The reason I did it like this here and not below is that I don't always need the `$client` (e.g. for getting the options), so in the other places I can avoid writing some code